packages/@sanity/form-builder/src/inputs/Array/styles/Array.css

```diff
@@ -23,6 +23,7 @@
 
 .movingItem {
   composes: movingItem from 'part:@sanity/base/theme/layout/helpers';
+  z-index: 100;
```

Comment:
If you use a zIndex just to get it over the current content: Use z-index in the current scope like z-index: 1 (if this is the only on current scope). 
If it is fixed or need to layer with other contents on the site, there are some global z-index (layers) to pick from. https://github.com/sanity-io/sanity/blob/master/packages/%40sanity/base/src/styles/variables/layers.css